### PR TITLE
you can now not dishonor your family

### LINKED ISSUE #29847: Sleeping carp users can wield and use flamethrowers
Round ID was 74532 I think

Just as the title said, I encountered an antag with sleeping carp, an atmos suit, and a flamethrower.
Managed to kill the entire station easily.
If I recall correctly Sleeping Carp is meant to stop the user from being able to use ranged weaponry.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #29847 
hopefully

### CODE REVIEW

**Comment by JJRcop** [comment]
Did you forget to commit something? this just adds a proc that isn't used

File: code/game/objects/items/weapons/flamethrower.dm [hunk 1]
```diff
@@ -252,3 +252,10 @@
 
 /obj/item/device/assembly/igniter/proc/ignite_turf(obj/item/weapon/flamethrower/F,turf/open/location,release_amount = 0.05)
 	F.default_ignite(location,release_amount)
+
+/obj/item/weapon/gun/proc/can_trigger_gun(var/mob/living/user)
+
+	!user.can_use_guns(src))
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/flamethrower.dm]: You're missing an `if` statement here

**Comment by PraiseRatvar** [comment]
@JJRcop /shrug/ i just found a fix which works for the issue mentioned on the comment

**Review by JJRcop — CHANGES_REQUESTED** [review]
This does nothing at all. (Not even compile) It tries to re-define an existing proc found in [code/modules/projectiles/gun.dm](https://github.com/tgstation/tgstation/blob/fedccea51931527a67405e14090a073bc2294a9d/code/modules/projectiles/gun.dm#L191), can_trigger_gun(). Flamethrowers don't even use can_trigger_gun, because they are not descendants of /obj/item/weapon/gun. Throw it all out and start over.

**Comment by MimicFaux** [comment]
Sleeping Carp Users can also be hit by syringe guns.

**Comment by ExcessiveUseOfCobblestone** [comment]
explain your post immediately

**Comment by PraiseRatvar** [comment]
**😅**

**Comment by Repukan** [comment]
They can also use pneumatic cannons IIRC

**Pull request closed without merging**